master/test/integration/api/api_trials_test.go

```diff
@@ -134,6 +162,135 @@ func trialDetailAPITests(
 
 }
 
+func trialProfilerMetricsTests(
+	t *testing.T, creds context.Context, cl apiv1.DeterminedClient, db *db.PgDB,
+) {
+	experiment := testutils.ExperimentModel()
+	err := db.AddExperiment(experiment)
+	assert.NilError(t, err, "failed to insert experiment")
+
+	trial := testutils.TrialModel(experiment.ID, testutils.WithTrialState(model.ActiveState))
+	err = db.AddTrial(trial)
+	assert.NilError(t, err, "failed to insert trial")
+
+	ctx, _ := context.WithTimeout(creds, time.Minute)
+	tlCl, err := cl.GetTrialProfilerMetrics(ctx, &apiv1.GetTrialProfilerMetricsRequest{
+		Labels: &trialv1.TrialProfilerMetricLabels{
+			TrialId:    int32(trial.ID),
+			Name:       "gpu_util",
+			AgentId:    "brad's agent",
+			MetricType: trialv1.TrialProfilerMetricLabels_PROFILER_METRIC_TYPE_SYSTEM,
+		},
+	})
+	assert.NilError(t, err, "failed to initiate trial profiler metrics stream")
+
+	for i := 0; i < 10; i++ {
+		match := randTrialProfilerSystemMetrics(trial.ID, "gpu_util", "brad's agent", "1")
+		_, err := cl.PostTrialProfilerMetricsBatch(creds, &apiv1.PostTrialProfilerMetricsBatchRequest{
+			Batch: match,
+		})
+		assert.NilError(t, err, "failed to insert mocked trial profiler metrics")
+
+		notMatch := randTrialProfilerSystemMetrics(trial.ID, "gpu_util", "someone else's agent", "1")
+		_, err = cl.PostTrialProfilerMetricsBatch(creds, &apiv1.PostTrialProfilerMetricsBatchRequest{
+			Batch: notMatch,
+		})
+		assert.NilError(t, err, "failed to insert mocked unmatched trial profiler metrics")
+
+		recvMetricsBatch, err := tlCl.Recv()
+		assert.NilError(t, err, "failed to stream metrics")
+
+		// Just nil the values since the floats and timestamps lose a little precision getting thrown
+		// around so much.
```

Comment:
I could probably do a better job at this, but somewhere from proto->go->postgres->go->proto timestamps are losing some precision. i can change it before merge to round to nearest 1/1000th or something.